tests/integration/cgroups.bats

```diff
@@ -78,3 +74,54 @@ EOF
         check_cgroup_value $CGROUP_MEMORY "memory.kmem.limit_in_bytes" 50331648
     fi
 }
+
+@test "runc create (no limits + no cgrouppath + no permission) succeeds" {
+    runc run -d --console-socket $CONSOLE_SOCKET test_cgroups_permissions
+    [ "$status" -eq 0 ]
+}
+
+@test "runc create (rootless + no limits + cgrouppath + no permission) fails with permission error" {
+    requires rootless
```

Comment:
I will think about this a little more when I get back from OSS, but I am not sure that we'd want the "fails with permission error" tests.

Reply:
Why wouldn't you want these? Test bloat?

Reply:
If it ends up working in the future (which may happen if systemd decides to enable the cgroupv2 delegation code), then the test will break. On the other hand, we could just include these and deal with that later.

Reply:
I can buy that. As long as getting a permission error isn't the desired behaviour right now (rather than say, some other error that might bubble up) and we don't want this test for regression, then I say kill the test and save test bloat/potential confusion later.

Reply:
There's a delegation section in the [cgroupv2 documentation](https://www.kernel.org/doc/Documentation/cgroup-v2.txt). Effectively you can set `-o nsdelegate` on the root mount of `cgroupv2` and then you get permissions to write to a sub-cgroup when you're in a cgroup namespace. I did a quick search for the actual patch but couldn't find it.